### ISSUE TITLE
ttnn.embedding_bw returns a rank-4 gradient for a rank-2 weight

### ISSUE THREAD

**Issue by timcober** [tenstorrent/tt-metal#53379]
**Component / Area**

ops

**Issue Type**

Bad Outputs

**Observed**

`ttnn.embedding_bw` returns a tensor of shape `{1, 1, num_embeddings, embedding_dim}` rank 4, for what is a rank-2 gradient. `spec_return_value` in `ttnn/cpp/ttnn/operations/embedding_backward/device/embedding_backward_device_operation.cpp` constructs the shape with two leading unit dimensions.

The weight the gradient must be accumulated into is rank 2, so every caller has to reshape. Under PyTorch/XLA the mismatch surfaces at the framework boundary, where the gradient is checked against the parameter it belongs to.

**Expected**

`{num_embeddings, embedding_dim}`, matching the weight tensor and the MLIR signature.

**1. Steps**

```python
import torch, ttnn

device = ttnn.open_device(device_id=0)
batch_size, seq_len, embedding_dim, num_embeddings = 2, 64, 160, 96

input_index = torch.randint(0, num_embeddings, (batch_size, seq_len))
input_tensor = ttnn.from_torch(input_index, dtype=ttnn.uint32, device=device)

weights_shape = (num_embeddings, embedding_dim)
weights = ttnn.from_torch(torch.randn(weights_shape), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

grad_shape = (1, 1, batch_size * seq_len, embedding_dim)
grad = ttnn.from_torch(torch.randn(grad_shape), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

out = ttnn.embedding_bw(input_tensor, weights, grad, dtype=ttnn.bfloat16)
print("weight shape:", weights_shape, " output shape:", out.shape)
```
```
weight shape: (96, 160)  output shape: Shape([1, 1, 96, 160])
```

After the one-line fix to remove 2 additional dimensions, the output is
`output shape: Shape([96, 160])`.

**2. Input data / link or description**

None required.

**3. Frequency**

Consistent. Every call; the shape is constructed unconditionally.
